### ISSUE TITLE
state file: Corrupted "<printer ...>" line gets system's name overwritten by printer name

### ISSUE THREAD

**Issue by tillkamppeter** [michaelrsweet/pappl#137]
The state file contains the following:
```
DNSSDName PostScript Printer Application
[...]
<Printer did="..." id="..." name="test" ...>
DNSSDName test
[...]
</Printer>
[...]
```
If everytjing goes normal, the file is correctly written and completely read (see #135) it works as expected.

Now, if the `<Printer ...>` line is corrupted (or not completely read as byte 4096 of the file is in the middle of it, see #135), and error is logged and reading of the file continued, but due to the broken `<Printer ...>` line not being considered, the following lines, representing the printer's configuration are considered part of the system's configuration, especially the `DNSSDName ...` line overwrites the name which was defined in the `DNSSDName ...` line at the very beginning of the file, making the system's name switching from the original name ("PostScript Printer Application") to the name of the corrupted printer ("test").

What needs to be done here is that if a "<Printer ...>" (or any "<XXX ...>") line is read and found that it is corrupt/incomplete/... and has to be ignored, everything following until "</Printer>" (or "</XXX>") has to get ignored.

**Comment by michaelrsweet** [comment]
@tillkamppeter Please attach a complete state file showing this problem.

**Comment by tillkamppeter** [comment]
I have observed this problem in a situation of issue #135, with 3 printers set up and the third not complete ly read back on restart. I assumed that this was caused by a corrupted `<Printer ...>` line and to try to reproduce I have now tested with this state file:
[ps-printer-app.state-broken-printer.txt](https://github.com/michaelrsweet/pappl/files/5832661/ps-printer-app.state-broken-printer.txt)
This file has only one printer and I have removed a part of the `<Printer ...>` line. Unfortunately, I cannot reproduce the bug with it and reviewing your code I can see that the corrupted line stops reading of the state file altogether and so nothing of the already read gets overwritten.
